pkg/daemon/daemon.go

```diff
@@ -2075,6 +2075,10 @@ func (dn *Daemon) validateOnDiskState(currentConfig *mcfgv1.MachineConfig) error
 	return nil
 }
 
+func canonizeImageURL(url string) string {
```

Comment:
This could use a doc comment (and also "canonicalize" is more canonical here) - [canonization is a different thing](https://en.wikipedia.org/wiki/Canonization) and I don't think this function quite qualifies for sainthood yet :smile: .  But even "canonicalize" would make someone wonder "canonicalize *how*?" I think.  

Further this is going to be a bit fragile because in theory we could (and in fact [have a TODO to change it](https://github.com/ostreedev/ostree-rs-ext/blob/332ad0dd799edbe023472a47c427ce81954b0fc5/lib/src/container/mod.rs#L240)).   So how about combining these two:  
```suggestion
// stripRegistryTransport removes the registry transport prefix, see https://docs.rs/ostree-ext/latest/ostree_ext/container/enum.Transport.html
func stripRegistryTransport(url string) string {
        return strings.Relpace(strings.Replace(url, "docker://", "", 1), "registry:", "", 1))
```
or so?

Could also use a unit test.

Reply:
Do we know that it will be always "docker://" or we should handle rest of the transport replacement as well that are supported by [OSTree](https://docs.rs/ostree-ext/latest/ostree_ext/container/enum.Transport.html) ?

Reply:
I think in OpenShift today we effectively hardcode `docker://` - i.e. we only support fetching from a registry in `osImageURL`.

So...yes.  Note https://github.com/openshift/os/pull/657 would actually switch things so that the initial state is from an `oci-archive`, but that's OK because it will today always be older anyways.

(A whole other problem here is that today we deploy as OCI, but when we later push the container image it gets converted to d2s2, which changes the digest...)

Reply:
yeah, makes sense.